File: chess.py/color_test.py
import unittest

from color import Color, color_row, other_color


class TestColor(unittest.TestCase):
    def test_other_color(self):
        self.assertEqual(other_color(Color.White), Color.Black)
        self.assertEqual(other_color(Color.Black), Color.White)

    def test_color_row(self):
        self.assertEqual(color_row(Color.White, 1), 1)
        self.assertEqual(color_row(Color.Black, 1), 8)

File: chess.ts/color.test.ts
import { expect, test } from '@jest/globals'
import { otherColor, colorRow } from './color'

test('otherColor', () => {
    expect(otherColor('White')).toBe('Black')
    expect(otherColor('Black')).toBe('White')
})

test('colorRow', () => {
    expect(colorRow('White',1)).toBe(1)
    expect(colorRow('White',2)).toBe(2)
    expect(colorRow('Black',1)).toBe(8)
    expect(colorRow('Black',2)).toBe(7)
})

